src/main/java/org/opensearch/geospatial/geojson/FeatureFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.geojson;
+
+
+import org.opensearch.geospatial.geojson.Feature.FeatureBuilder;
+
+import java.util.Map;
+
+/**
+ * FeatureFactory helps to create {@link Feature} instance based on user input
+ */
+public class FeatureFactory {
+
+    /**
+     * The factory method to create an instance of Feature from input, and, throws {@link IllegalArgumentException} for invalid input.
+     *
+     * @param input the object from where {@link Feature} will be extracted
+     * @return Feature Instance from input
+     * @throws IllegalArgumentException if invalid input is provided
+     */
+    public static Feature create(Map<String, Object> input) throws IllegalArgumentException {
```

Comment:
Why IllegalArgumentException is defined in the method signature, it's unchecked exception?